tests/unit/s2n_client_psk_extension_test.c

```diff
@@ -1543,5 +1543,121 @@ int main(int argc, char **argv)
         EXPECT_SUCCESS(s2n_io_pair_close(&io_pair));
     }
 
+    /**
+     * Ensure obfuscated_ticket_age and binder values are updated on a client hello after a HRR
+     *
+     *= https://tools.ietf.org/rfc/rfc8446#section-4.1.2
+     *= type=test
+     *# -   Updating the "pre_shared_key" extension if present by recomputing
+     *#     the "obfuscated_ticket_age" and binder values and (optionally)
+     *#     removing any PSKs which are incompatible with the server's
+     *#     indicated cipher suite.
+     **/
+    {
+        DEFER_CLEANUP(struct s2n_cert_chain_and_key *chain_and_key,
+                      s2n_cert_chain_and_key_ptr_free);
+        EXPECT_SUCCESS(s2n_test_cert_chain_and_key_new(&chain_and_key,
+                                                       S2N_DEFAULT_ECDSA_TEST_CERT_CHAIN,
+                                                       S2N_DEFAULT_ECDSA_TEST_PRIVATE_KEY));
+
+        DEFER_CLEANUP(struct s2n_config *config = s2n_config_new(),
+                      s2n_config_ptr_free);
+        EXPECT_SUCCESS(s2n_config_add_cert_chain_and_key_to_store(config, chain_and_key));
+        EXPECT_SUCCESS(s2n_config_set_cipher_preferences(config, "default_tls13"));
+        EXPECT_SUCCESS(s2n_config_disable_x509_verification(config));
+        EXPECT_SUCCESS(s2n_setup_ticket_key(config));
+
+        DEFER_CLEANUP(struct s2n_connection *server_conn = s2n_connection_new(S2N_SERVER),
+                      s2n_connection_ptr_free);
+        EXPECT_NOT_NULL(server_conn);
+        EXPECT_SUCCESS(s2n_connection_set_config(server_conn, config));
+
+        DEFER_CLEANUP(struct s2n_connection *client_conn = s2n_connection_new(S2N_CLIENT),
+                      s2n_connection_ptr_free);
+        EXPECT_NOT_NULL(client_conn);
+        EXPECT_SUCCESS(s2n_connection_set_config(client_conn, config));
+
+        struct s2n_test_io_pair io_pair = { 0 };
+        EXPECT_SUCCESS(s2n_io_pair_init_non_blocking(&io_pair));
+        EXPECT_SUCCESS(s2n_connections_set_io_pair(client_conn, server_conn, &io_pair));
+
+        /* Force the HRR path */
+        client_conn->security_policy_override = &security_policy_test_tls13_retry;
+
+        client_conn->actual_protocol_version = S2N_TLS13;
+        client_conn->secure.cipher_suite = &s2n_tls13_aes_128_gcm_sha256;
+
+        /* Set a resumption psk */
```

Comment:
So, I've been thinking over this test and I think we can make it better. What worries me about how it currently works is that we are looking for obfuscated ticket age and the binders by skipping through the client hello. The issue with this technique is that if we get the skip values wrong at all, the test will still pass probably.
I think we can do it a little better by using the fact that psk identities are sent in the clear. So knowing that we can search the client hello for the identity using s2n_stuffer_skip_read_until(), and then once we have the identity the obfuscated_ticket_age is right after that, and the binders list is right after that.
This is a much easier way of finding these values and it's easier for reviewers to understand what is happening. 
Also can switch to using external psks so that the identity is easier to understand. [This is how you set up an external psk](https://github.com/aws/s2n-tls/blob/main/tests/unit/s2n_psk_test.c#L728).

Reply:
Your concern definitely makes sense, and seeking the stuffers relative to the psk identity definitely seems nicer. I changed this in 70a4854442d416b2f30cbad74035dae8b6ceb397.

The resumption psk was kept instead of an external psk due to setting obfuscated ticket age values to 0 for external psks:
https://github.com/aws/s2n-tls/blob/c920a3e0e6149694e2f5ddd49ec053cabca56ec2/tls/extensions/s2n_client_psk.c#L113-L116